src/libraries/Microsoft.PowerFx.Core/Syntax/NamedFormula.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Parser;
+using Microsoft.PowerFx.Core.Syntax.Nodes;
+using Microsoft.PowerFx.Core.Utils;
+
+namespace Microsoft.PowerFx.Core.Syntax
+{
+    /// <summary>
+    /// This encapsulates a named formula: its original script, the parsed result, and any parse errors.
+    /// </summary>
+    internal class NamedFormula
```

Comment:
>NamedFormula [](http://example.com/codeflow?start=19&length=12)

Is this class mutable or not? 
See slide 57 in greg's XR - what happens when we have dedicated UI and show each formula as its own UI element (Rather than a single giant Script  value)